Error when Compiling the DCU version

### Describe the bug

I try to compile the latest abacus on Sugon dcu, and get the bellow error:
```
source/module_hsolver/CMakeFiles/hsolver.dir/diago_dav_subspace.cpp.o: In function `hsolver::Diago_DavSubspace<std::complex<float>, psi::DEVICE_GPU>::diag_zhegvx(int const&, int const&, std::complex<float>*, std::complex<float>*, int const&, float*, std::complex<float>*, bool, bool)':
/public/home/abacus/abacus-dcu/source/module_hsolver/diago_dav_subspace.cpp:600: undefined reference to `hsolver::dngvx_op<std::complex<float>, psi::DEVICE_GPU>::operator()(psi::DEVICE_GPU const*, int, int, std::complex<float>*, std::complex<float>*, int, float*, std::complex<float>*)'
source/module_hsolver/CMakeFiles/hsolver.dir/diago_dav_subspace.cpp.o: In function `hsolver::Diago_DavSubspace<std::complex<double>, psi::DEVICE_GPU>::diag_zhegvx(int const&, int const&, std::complex<double>*, std::complex<double>*, int const&, double*, std::complex<double>*, bool, bool)':
/public/home/abacus/abacus-dcu/source/module_hsolver/diago_dav_subspace.cpp:600: undefined reference to `hsolver::dngvx_op<std::complex<double>, psi::DEVICE_GPU>::operator()(psi::DEVICE_GPU const*, int, int, std::complex<double>*, std::complex<double>*, int, double*, std::complex<double>*)'
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/abacus_pw.dir/build.make:712: abacus_pw] Error 1
make[1]: *** [CMakeFiles/Makefile2:791: CMakeFiles/abacus_pw.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/source/module_hsolver/kernels/rocm/dngvd_op.hip.cu
+++ b/source/module_hsolver/kernels/rocm/dngvd_op.hip.cu
@@ -137,4 +137,46 @@ void dnevx_op<std::complex<double>, psi::DEVICE_GPU>::operator()(const psi::DEVI
     hipErrcheck(hipMemcpy(_eigenvalue, eigenvalue.data(), sizeof(double) * eigenvalue.size(), hipMemcpyHostToDevice));
 }
 
+
+template <>
+void dngvx_op<std::complex<float>, psi::DEVICE_GPU>::operator()(
+                const psi::DEVICE_GPU* d,
+                const int nbase,
+                const int ldh,
+                std::complex<float>* hcc,
+                std::complex<float>* scc,
+                const int m,
+                float* eigenvalue,
+                std::complex<float>* vcc)
+{
+}
+
+template <>
+void dngvx_op<std::complex<double>, psi::DEVICE_GPU>::operator()(
+                const psi::DEVICE_GPU* d,
+                const int nbase,
+                const int ldh,
+                std::complex<double>* hcc,
+                std::complex<double>* scc,
+                const int m,
+                double* eigenvalue,
+                std::complex<double>* vcc)
+{
+}
+
+#ifdef __LCAO
+template <>
+void dngvx_op<double, psi::DEVICE_GPU>::operator()(
+                const psi::DEVICE_GPU* d,
+                const int nbase,
+                const int ldh,
+                double* hcc,
+                double* scc,
+                const int m,
+                double* eigenvalue,
+                double* vcc)
+{
+}
+#endif // __LCAO
+
 } // namespace hsolver

